app/src/sharedTest/java/org/oppia/android/app/databinding/StateAssemblerMarginBindingAdaptersTest.kt

```diff
@@ -0,0 +1,455 @@
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.platformparameter.PlatformParameterModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.TestImageLoaderModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TOLERANCE = 1e-5f
+
+/** Tests for [StateAssemblerMarginBindingAdapters]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = StateAssemblerMarginBindingAdaptersTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class StateAssemblerMarginBindingAdaptersTest {
+
+  @Inject
+  lateinit var context: Context
+
+  @get:Rule
+  var activityRule: ActivityScenarioRule<StateAssemblerMarginBindingAdaptersTestActivity> =
+    ActivityScenarioRule(
+      Intent(
+        ApplicationProvider.getApplicationContext(),
+        StateAssemblerMarginBindingAdaptersTestActivity::class.java
+      )
+    )
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    Intents.init()
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  fun testMarginBindingAdapters_explorationViewMargin_ltrIsEnabled_marginsForLtrIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setExplorationViewMargins(textView)
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      getExplorationMarginStart()
+    )
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      getExplorationMarginEnd()
+    )
+  }
+
+  @Test
+  fun testMarginBindingAdapters_explorationViewMargin_rtlIsEnabled_marginsForRtlIsCorrect() {
```

Comment:
This test is a copy of the other 4 test cases, the other four test cases having annotations of portland and both for tablet port/land too, Is it necessary to have the same test cases 5 times? Any specific reason or can we have one test case and depend on it that it will pass on all the configuration as currently, it's also the same as all 5 test cases are same.

Reply:
Nice catch. But it has to test for "ExplorationViewMargin" which was missing. I have changed this test to `setExplorationViewMargin` and other tests are for setExplorationSplitViewMargin().

Reply:
Ah, no actually this case has to test for `setExplorationViewMargin` and not `setExplorationSplitViewMargin` I corrected this please check.